Replace thread gem with concurrent-ruby gem

annaagaf opened the pull request:
This gem depends on depends on gem [thread](https://rubygems.org/gems/thread/versions/0.2.2) which is licensed under WTFPL license. 

That gem is not actively maintained and all of its functionality is included into a well-known gem [concurrent-ruby](https://github.com/ruby-concurrency/concurrent-ruby), which is distributed under MIT license.

Deleting Gemfile.lock, because as it is ignored when installing the gem, the presense of Gemfile.lock is more likely to hide issues with incompatibility with newer versions of dependencies that applications might use.

take-five commented on .gitignore:
```diff
@@ -3,3 +3,6 @@ pkg
 .tags
 .tags1
 coverage
+.idea/
```
Since using IDEA is your personal preference, this change shouldn't be checked into the repository.
I recommend you to configure it in your global `.gitignore`. Here's a good article about how to do it: https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/

annaagaf replied:
Good point, done.

annaagaf commented:
@take-five: Added more context and reasoning to all commit messages.
@indrekj: Changed to minor version bump, good point about backwards compatibility. Regarding Gemfile.lock, I would argue that gem build breaking would be preferable to application build breaking due to incompatible dependencies - however, the point about rubocop makes sense; restored Gemfile.lock to comply to the agreed convention.

take-five commented:
- Commits [`c07f97e` (#78)](https://github.com/salemove/freddy/pull/78/commits/c07f97e5716a32fe19b6aab7c455e1cebc14f33e) and [`c151e3e` (#78)](https://github.com/salemove/freddy/pull/78/commits/c151e3e1ecd5489ca5b6ca6237bc9a4c0f92bc31) can be squashed together
- Commits [`246379e` (#78)](https://github.com/salemove/freddy/pull/78/commits/246379e9b78cb011ced8ecec246f6f5a66a63c82) and [`cb5bcbd` (#78)](https://github.com/salemove/freddy/pull/78/commits/cb5bcbdfe757c61d25b1b2e154efcfb9c8066931) can be squashed together
- The .idea change should be edited out of the first commit and then commit [`7b25fc8` (#78)](https://github.com/salemove/freddy/pull/78/commits/7b25fc801423710a3203a5755d595d4e5ad46a02) is not needed anymore

CI fails for some reason

indrekj reviewed (approved):
No need for `-rc.1`, but otherwise looks good I think

annaagaf commented:
@indrekj Reasoning for rc-1 is that the gem has to be published before being actually auto-tested in applications, so the idea is to update to stable after validation is successful.

take-five reviewed (approved):
Typo in commit
> As only an internal library is changed and changes are backwards-compatible, updating **major** version.

You meant _minor_?
Otherwise LGTM

take-five commented on freddy.gemspec:
```diff
@@ -12,19 +12,19 @@ Gem::Specification.new do |spec|
   spec.license       = 'MIT'
   spec.homepage      = 'https://github.com/salemove/freddy'
   spec.required_ruby_version = '>= 2.7'
+  spec.metadata['rubygems_mfa_required'] = 'true'
```
This line seems to be preventing GH-action from publishing the gem

annaagaf replied:
this is rubocop requirement. I can rollback rubocop changes to auto-publish, but then it probably will need to be addressed later?
it also seems to be enabled in jaeger-client-ruby, can check how publishing is solved there.

annaagaf replied:
there's just no publish action there, instead, there is instruction on manual publishing in readme.

take-five replied:
> can check how publishing is solved there

There's no auto-publishing in there
https://github.com/salemove/jaeger-client-ruby/blob/master/.github/workflows/ci.yml

I think we can disable this Rubocop rule? @indrekj WDYT?

indrekj replied:
I think so
